test/test_torch.py

```diff
@@ -18073,6 +18073,30 @@ def compare_with_numpy(contiguous_input=True):
         self.assertEqual(a[5:].real, a.real[5:])
         self.assertEqual(a[5:].imag, a.imag[5:])
 
+    @onlyOnCPUAndCUDA
+    @dtypes(*torch.testing.get_all_complex_dtypes())
+    def test_set_real_imag(self, device, dtype):
+        x = torch.randn(10, dtype=dtype)
+
+        for d in torch.testing.get_all_dtypes():
```

Comment:
You can use

```
@dtypes(*product(torch.testing.get_all_complex_dtypes(), torch.testing.get_all_dtypes()))
```

and the third argument in your signature will be `dtypes` and it will be an element from the product (a tuple of two elements where dtypes[0] is from torch.testing.get_all_complex_dtypes() and dtypes[1] is from torch.testing.get_all_dtypes()).